cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/NamespaceHttpHandler.java

```diff
@@ -164,9 +161,17 @@ public void create(HttpRequest request, HttpResponder responder, @PathParam("nam
     }
   }
 
+  /**
+   * DO NOT DOCUMENT THIS API
+   */
   @DELETE
-  @Path("/namespaces/{namespace-id}")
+  @Path("/unrecoverable/namespaces/{namespace-id}")
   public void delete(HttpRequest request, HttpResponder responder, @PathParam("namespace-id") String namespace) {
+    // NOTE: DO NOT DOCUMENT
+    if (!cConf.getBoolean(Constants.Dangerous.UNRECOVERABLE_RESET, Constants.Dangerous.DEFAULT_UNRECOVERABLE_RESET)) {
+      responder.sendStatus(HttpResponseStatus.FORBIDDEN);
+      return;
+    }
     if (isReserved(namespace)) {
```

Comment:
So we cannot delete all apps, streams etc in default namespace? I thought the behavior should be to delete everything and not the namespace/database in hbase and explore